rosette/src/Table.cc

```diff
@@ -305,9 +308,11 @@ void RblTable::linearCheckSize() {
     }
 }
 
-
+// We need to keep track of the number of deleted entries separately
+// because until the table is rehashed, they still occupy space in the hash table.
 void RblTable::hashCheckSize() {
-    if (4 * numberOfEntries > 3 * maxEntries) {  // i.e., the table is 75% full
+    if (4 * (numberOfEntries+numberOfDeletedHashEntries) > 3 * maxEntries) {  // i.e., the table is
+        // 75% full
         hashResize();
```

Comment:
I don't see any changes to hashResize()

Shouldn't it now reset numberOfDeletedHashEntries to 0?

Reply:
Yes that is what https://github.com/rchain/rchain/pull/265/files#diff-df763b27e057305f90357fcdb04c46b4R169 does. hashResize calls rehashCompletely which sets numberOfDeletedHashEntries to 0.